dart/lib/src/event_processor/enricher/io_enricher_event_processor.dart

```diff
@@ -54,8 +54,21 @@ class IoEnricherEventProcessor implements EnricherEventProcessor {
   List<SentryRuntime> _getRuntimes(List<SentryRuntime>? runtimes) {
     // Pure Dart doesn't have specific runtimes per build mode
     // like Flutter: https://flutter.dev/docs/testing/build-modes
-    final dartRuntime = SentryRuntime(
+
+    // Extracts the semantic version and channel from the full version string.
+    // 3.5.0-180.3.beta (beta) (Wed Jun 5 15:06:15 2024 +0000) on "android_arm64"
+    // turns into 3.5.0-180.3.beta (beta)
+    String version = Platform.version;
+    RegExp channelRegex = RegExp(r'\((stable|beta|dev)\)');
```

Comment:
constructing & executing a RegExp may be relatively costly and this doesn't need to run on every event. Maybe consider extracting `dartRuntime` to a `late` variable.

Reply:
yup makes sense! thx